src/Mod/Sketcher/Gui/SoZoomTranslation.cpp

```diff
@@ -66,16 +66,16 @@ void SoZoomTranslation::initClass()
     SO_ENABLE(SoGetMatrixAction, SoViewVolumeElement);
 }
 
-float SoZoomTranslation::getScaleFactor(SoAction* action) const
+float SoZoomTranslation::calculateScaleFactor(SoAction* action) const
 {
     // Dividing by 5 seems to work well
     SbViewVolume vv = SoViewVolumeElement::get(action->getState());
     float aspectRatio = SoViewportRegionElement::get(action->getState()).getViewportAspectRatio();
-    float scale = vv.getWorldToScreenScale(SbVec3f(0.f, 0.f, 0.f), 0.1f) / (5*aspectRatio);
-    return scale;
+    scaleFactor = vv.getWorldToScreenScale(SbVec3f(0.f, 0.f, 0.f), 0.1f) / (5*aspectRatio);
+    return scaleFactor;
 }
 
-SoZoomTranslation::SoZoomTranslation()
+SoZoomTranslation::SoZoomTranslation(): scaleFactor(0)
```

Comment:
Default 0 for `scaleFactor` might not be the best idea, especially if there is division involved. 1 sounds better, depending on what the definition is.

Reply:
The objective is to store whatever the ZoomTranslation calculates. 

Retrieving the scaleFactor before (internal) calculation will always result in a wrong value. This means that 1 will be as wrong as 0.

There is no risk of exception as the the scale factor is multiplied to a relative translation.

Then, perhaps 0 best serves a user of the client code to realise the mistake, if trying to retrieve a scale before the node is actually in running in the scenegraph.